common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -828,19 +827,27 @@ public Object run() {
                 // up to the next power of two and so will overflow otherwise.
                 final int capacity =
                         Math.max(Math.min(maxCapacity, MAX_ALLOWED_MPSC_CAPACITY), MIN_MAX_MPSC_CAPACITY);
-                return new MpscChunkedArrayQueue<T>(MPSC_CHUNK_SIZE, capacity, true);
-            } else {
-                return new MpscLinkedAtomicQueue<T>();
+                return new MpscChunkedArrayQueue<T>(MPSC_CHUNK_SIZE, capacity);
             }
+            // TODO(scott): we don't respect capacity because there is no bounded atomic queue implementation.
+            return new MpscLinkedAtomicQueue<T>();
+        }
+
+        static <T> Queue<T> newMpscQueue() {
+            // TODO(scott): there is no unbounded array based queue. Currently
```

Comment:
@nitsanw - FYI. I didn't use `MpscLinkedQueue` in this PR for GC concerns. I have not done any evaluation ... we can do this as a followup PR if necessary. I did want to upgrade to the latest version of JCTools though.